pkg/vdr/web/resolver_test.go

```diff
@@ -0,0 +1,112 @@
+	validDIDWithPath = prefix + "www.example.org:user:example"
+
+	invalidDIDNoMethod = "did:" + validURL
+	invalidDIDNoPrefix = validURL
+
+	validDoc = `{
+  		"@context": ["https://w3id.org/did/v1"],
+  		"id": "did:web:www.example.org"
+	}`
+
+	invalidDoc = `{}`
+)
+
+func TestParseDID(t *testing.T) {
+	t.Run("test parse did success", func(t *testing.T) {
+		url, err := parseDIDWeb(validDID)
+		require.NoError(t, err)
+		require.Equal(t, "https://"+validURL+defaultPath, url)
+		url, err = parseDIDWeb(validDIDWithPath)
+		require.NoError(t, err)
+		require.Equal(t, "https://"+validURLWithPath+documentPath, url)
+	})
+
+	t.Run("test parse did failure", func(t *testing.T) {
+		v := New()
+		doc, err := v.Read(invalidDIDNoMethod)
+		require.Error(t, err)
+		require.Nil(t, doc)
+		require.Contains(t, err.Error(), "does not conform to generic did standard")
+		doc, err = v.Read(invalidDIDNoPrefix)
+		require.Error(t, err)
+		require.Nil(t, doc)
+		require.Contains(t, err.Error(), "does not conform to generic did standard")
+	})
+}
+
+func TestResolveDID(t *testing.T) {
+	t.Run("test resolve did with request failure", func(t *testing.T) {
+		s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(invalidDoc))
+			require.NoError(t, err)
+		}))
+		defer s.Close()
+		did := fmt.Sprintf("did:web:%s", strings.TrimPrefix(s.URL, "https://"))
+		v := New()
+		doc, err := v.Read(did)
+		require.Nil(t, doc)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "http request unsuccessful")
+	})
+	t.Run("test resolve did with invalid doc format failure", func(t *testing.T) {
+		s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(invalidDoc))
+			require.NoError(t, err)
+		}))
+		defer s.Close()
+		did := fmt.Sprintf("did:web:%s", strings.TrimPrefix(s.URL, "https://"))
+		v := New()
+		doc, err := v.Read(did, vdr.WithHTTPClient(s.Client()))
+		require.Nil(t, doc)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "error parsing did doc")
+	})
+	t.Run("test resolve did success", func(t *testing.T) {
+		s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(validDoc))
+			require.NoError(t, err)
+		}))
+		defer s.Close()
+		did := fmt.Sprintf("did:web:%s", strings.TrimPrefix(s.URL, "https://"))
+		v := New()
+		_, err := v.Read(did, vdr.WithHTTPClient(s.Client()))
+		require.Nil(t, err)
+	})
+	t.Run("test resolve did with path success", func(t *testing.T) {
+		s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(validDoc))
+			require.NoError(t, err)
+		}))
+		defer s.Close()
+		did := fmt.Sprintf("did:web:%s:user:johnsmith", strings.TrimPrefix(s.URL, "https://"))
+		v := New()
+		_, err := v.Read(did, vdr.WithHTTPClient(s.Client()))
+		require.Nil(t, err)
```

Comment:
you should check document as well, not just error

Reply:
`doc, err := v.Read(didID, vdr.WithHTTPClient(s.Client()))
		require.Nil(t, err)
		expectedDoc, err := did.ParseDocument([]byte(validDoc))
		require.Nil(t, err)
		require.Equal(t, expectedDoc, doc)`
does this look good? (given validDoc is what's being returned by server)